src/go/k8s/apis/redpanda/v1alpha1/console_types.go

```diff
@@ -0,0 +1,189 @@
+	ServeFrontend bool `json:"serveFrontend"`
+
+	// +optional
+	Server Server `json:"server"`
+
+	Schema        Schema                 `json:"schema"`
+	ClusterKeyRef corev1.ObjectReference `json:"clusterKeyRef"`
+	Deployment    Deployment             `json:"deployment"`
+	Connect       Connect                `json:"connect"`
+}
+
+// Server is the Console app HTTP server config
+type Server struct {
+	// +kubebuilder:default="30s"
+	ServerGracefulShutdownTimeout string `json:"gracefulShutdownTimeout,omitempty" yaml:"gracefulShutdownTimeout,omitempty"`
+
+	HTTPListenAddress string `json:"listenAddress,omitempty" yaml:"listenAddress,omitempty"`
+
+	// +kubebuilder:default=8080
+	HTTPListenPort int `json:"listenPort,omitempty" yaml:"listenPort,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerReadTimeout string `json:"readTimeout,omitempty" yaml:"readTimeout,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerWriteTimeout string `json:"writeTimeout,omitempty" yaml:"writeTimeout,omitempty"`
+
+	// +kubebuilder:default="30s"
+	HTTPServerIdleTimeout string `json:"idleTimeout,omitempty" yaml:"idleTimeout,omitempty"`
+
+	// +kubebuilder:default=4
+	CompressionLevel int `json:"compressionLevel,omitempty" yaml:"compressionLevel,omitempty"`
+
+	BasePath string `json:"basePath,omitempty" yaml:"basePath,omitempty"`
+
+	// +kubebuilder:default=true
+	SetBasePathFromXForwardedPrefix bool `json:"setBasePathFromXForwardedPrefix,omitempty" yaml:"setBasePathFromXForwardedPrefix,omitempty"`
+
+	// +kubebuilder:default=true
+	StripPrefix bool `json:"stripPrefix,omitempty" yaml:"stripPrefix,omitempty"`
+}
+
+// Schema defines configurable fields for Schema Registry
+type Schema struct {
+	Enabled bool `json:"enabled"`
+}
+
+// Deployment defines configurable fields for the Console Deployment resource
+type Deployment struct {
+	Image string `json:"image"`
+
+	// +kubebuilder:default=1
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// +kubebuilder:default=0
+	MaxUnavailable int32 `json:"maxUnavailable,omitempty"`
+
+	// +kubebuilder:default=1
+	MaxSurge int32 `json:"maxSurge,omitempty"`
+}
+
+// Connect defines configurable fields for Kafka Connect
+type Connect struct {
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="15s"
+	ConnectTimeout *metav1.Duration `json:"connectTimeout,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="60s"
+	ReadTimeout *metav1.Duration `json:"readTimeout,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="6s"
+	RequestTimeout *metav1.Duration `json:"requestTimeout,omitempty"`
+
+	Clusters []ConnectCluster `json:"clusters,omitempty"`
+}
+
+// ConnectCluster defines configurable fields for the Kafka Connect cluster
+type ConnectCluster struct {
```

Comment:
can you have someone from kafka connect team take a look at this? I am not that familiar with it

Reply:
cc @tomasz-sadura @bochenekmartin